js/main.js

```diff
@@ -156,28 +169,78 @@ function saveVerse(event) {
   var $verseDiv = document.createElement('div');
   var $verseP = document.createElement('p');
   var $entryDiv = document.createElement('div');
-  var $entryP = document.createElement('p');
+  var $innerEntryDiv = document.createElement('div');
+  var $innerButtonDiv = document.createElement('div');
+  var $newEntryButton = document.createElement('button');
 
-  $ul.appendChild($li);
+  $ul.prepend($li);
   $li.appendChild($verseDiv);
   $li.appendChild($entryDiv);
   $verseDiv.appendChild($verseP);
-  $entryDiv.appendChild($entryP);
+  $entryDiv.appendChild($innerEntryDiv);
+  $entryDiv.appendChild($innerButtonDiv);
+  $innerButtonDiv.appendChild($newEntryButton);
 
   $li.className = 'desktop-display-flex';
   $verseDiv.className = 'column-half';
   $verseP.className = 'saved-verse-box';
-  $entryDiv.className = 'column-half';
-  $entryP.className = 'padding-around;';
+  $entryDiv.className = 'column-half margin-auto';
+  $innerEntryDiv.id = 'saved-journals';
+  $newEntryButton.className = 'margin-auto bold cursor-pointer black-button padding-around';
+  $newEntryButton.textContent = 'Edit';
+  $newEntryButton.id = 'entry' + data.nextEntryId;
 
   var $verse = document.getElementById('verse').textContent;
   var $verseTitle = document.getElementById('verse-title').textContent;
   $verseP.textContent = $verse + ' - ' + $verseTitle;
 
+  data.editing = document.getElementById($newEntryButton.id);
+
   var entry = {};
   entry.verse = $verseP.textContent;
+  entry.title = $verseTitle;
+  entry.ID = data.nextEntryId;
+  data.nextEntryId++;
   data.entries.push(entry);
 
   data.view = 'journal-page';
   viewSwapper();
 }
+
+document.addEventListener('click', function (event) {
+  data.view = 'edit-page';
+  var $ul = document.querySelector('#journal-entries');
+  $ul.className = 'padding-bottom';
+  if (event.target.textContent === 'Edit') {
+    viewSwapper();
+    if ($ul.hasChildNodes() === false) {
+      var $li = document.createElement('li');
+      var $p = document.createElement('p');
+      $p.id = 'journal-verse';
+
+      $ul.appendChild($li);
+      $li.appendChild($p);
+      $li.id = data.nextEntryId - 1;
+      $p.textContent = data.entries[data.nextEntryId - 1].verse;
+    } else {
+      var $pVerse = document.getElementById('journal-verse');
+      $pVerse.textContent = data.entries[data.nextEntryId - 1].verse;
```

Comment:
Using `data.nextEntryId` to determine the length of the array is a bad idea, if you ever add the ability to delete entries, then the next id will not match the actual count of items stored.